internal/services/containerapps/container_app_resource.go

```diff
@@ -0,0 +1,414 @@
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil || existing.Model == nil {
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			model := existing.Model
+
+			// Delta-updates need the secrets back from the list API, or we'll end up removing them or erroring out.
+			secretsResp, err := client.ListSecrets(ctx, *id)
+			if err != nil || secretsResp.Model == nil {
+				if secretsResp.HttpResponse == nil || secretsResp.HttpResponse.StatusCode != http.StatusNoContent {
+					return fmt.Errorf("retrieving secrets for update for %s: %+v", *id, err)
+				}
+			}
+			model.Properties.Configuration.Secrets = helpers.UnpackContainerSecretsCollection(secretsResp.Model)
+
+			if metadata.ResourceData.HasChange("revision_mode") {
+				revisionMode := containerapps.ActiveRevisionsMode(state.RevisionMode)
+				model.Properties.Configuration.ActiveRevisionsMode = &revisionMode
+			}
+
+			if metadata.ResourceData.HasChange("ingress") {
+				model.Properties.Configuration.Ingress = helpers.ExpandContainerAppIngress(state.Ingress, id.ContainerAppName)
+			}
+
+			if metadata.ResourceData.HasChange("registry") {
+				model.Properties.Configuration.Registries = helpers.ExpandContainerAppRegistries(state.Registries)
+
+			}
+
+			if metadata.ResourceData.HasChange("dapr") {
+				model.Properties.Configuration.Dapr = helpers.ExpandContainerAppDapr(state.Dapr)
+
+			}
+
+			if metadata.ResourceData.HasChange("template") {
+				allProbesRemoved := helpers.ContainerAppProbesRemoved(metadata)
+				if allProbesRemoved {
+					nilProbes := make([]containerapps.ContainerAppProbe, 0)
+					containers := *model.Properties.Template.Containers
+					containers[0].Probes = &nilProbes
+					model.Properties.Template.Containers = &containers
+				}
+			}
+
+			if metadata.ResourceData.HasChange("secret") {
+				model.Properties.Configuration.Secrets = helpers.ExpandContainerSecrets(state.Secrets)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = tags.Expand(state.Tags)
+			}
+
+			model.Properties.Template = helpers.ExpandContainerAppTemplate(state.Template, metadata)
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ContainerAppResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			if metadata.ResourceDiff != nil && metadata.ResourceDiff.HasChange("secret") {
+				stateSecretsRaw, configSecretsRaw := metadata.ResourceDiff.GetChange("secret")
+				stateSecrets := stateSecretsRaw.(*schema.Set).List()
+				configSecrets := configSecretsRaw.(*schema.Set).List()
+				// Check there's not less
+				if len(configSecrets) < len(stateSecrets) {
+					return fmt.Errorf("cannot remove secrets from Container Apps at this time. Please see `https://github.com/microsoft/azure-container-apps/issues/395` for more details")
```

Comment:
we should probably be more explicit this is an API limitation in the error?

```suggestion
					return fmt.Errorf("cannot remove secrets from Container Apps at this time due to a limitation in the Azure API. Please see `https://github.com/microsoft/azure-container-apps/issues/395` for more details")
```